src/Cassandra/RequestTrackingInfo.cs

```diff
@@ -0,0 +1,32 @@
+﻿//
+//      Copyright (C) DataStax Inc.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+using System.Collections.Concurrent;
+
+namespace Cassandra
+{
+    public class RequestTrackingInfo
+    {
+        public RequestTrackingInfo()
+        {
+            this.Items = new ConcurrentDictionary<string, object>();
+        }
+
+        public ConcurrentDictionary<string, object> Items { get; }
+
+        public IStatement Statement { get; set; }
```

Comment:
Do we need a public `setter` here? Ideally the driver should set this when it creates this object and these properties should be immutable (well the `Items` dictionary can be modified but the properties themselves shouldn't).

Reply:
I changed it so the statement is passed by constructor and the public property is getter only